File: examples/strongscript/16.ts
class B {
	foo():!number { return 2 }
}
class C {
	foo():any { return "hello" }
}

class D {
	b:!B
	c:!C
	main():!number {
		this.c = new C()
		this.b = <!B>this.c
		return this.b.foo()
	}
}

File: examples/reticulated_python/16.py
class B:
	def foo() -> Int:
		return 2

class C:
	def foo() -> Dyn:
		return "hello"

@fields({'b' : B, 'c' : C})
class C:
	b = B()
	c = C()
	def main(self:C) -> Int:
		self.c = C()
		self.b = self.c
		return self.b.foo()


print(C().main())
